test/e2e/source_api_server_test.go

```diff
@@ -126,22 +126,24 @@ func TestApiServerSource(t *testing.T) {
 	client := setup(t, true)
 	defer tearDown(client)
 
-	// creates ServiceAccount and ClusterRoleBinding with default cluster-admin role
-	cr := resources.ClusterRole(clusterRoleName,
-		resources.WithRuleForClusterRole(&rbacv1.PolicyRule{
-			APIGroups: []string{rbacv1.APIGroupAll},
+	// creates ServiceAccount and RoleBinding with a role for reading pods and events
+	r := resources.Role(roleName,
+		resources.WithRuleForRole(&rbacv1.PolicyRule{
+			APIGroups: []string{""},
 			Resources: []string{"events"},
```

Comment:
```suggestion
			Resources: []string{"events", "pods"},
```

Then the second WithRuleForRole can be removed.

Reply:
updated, thanks!